### PR TITLE
build: Refactor ceedling files to avoid repetition

### PR DESCRIPTION
**Pull request by expressvpn-raihaan-m**
## Description
Refactor repeated ceedling configs

## Motivation and Context
Makes it easier to bump version numbers of dependencies or add shared configs

## How Has This Been Tested?
CI Tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All active GitHub checks are passing  
- [ ] The correct base branch is being used, if not `main`

### CODE REVIEW

**Review by expressvpn-tom-l — COMMENTED** [review]
Nice improvement! Thanks @expressvpn-raihaan-m 

My only concern is whether the configuration settings AND build definitions are still exactly the same as before. I couldn't spot any problems on mac/linux/ios/android platform, but I'm not so sure about Windows.

File: windows/wolfssl-user_settings-64.h [hunk 1]
```diff
@@ -6,11 +6,6 @@
 #error This user_settings.h header is only designed for Windows
 #endif
 
-/* Check WOLFSSL_DTLS_ALLOW_FUTURE flag */
-#ifndef WOLFSSL_DTLS_ALLOW_FUTURE
-#error WOLFSSL_DTLS_ALLOW_FUTURE is not defined
-#endif
```
[the remark this answers is not in the document]
Comment (expressvpn-raihaan-m) [reply windows/wolfssl-user_settings-64.h:12]: We are removing this define, its no longer valid. I've added a note in the commit about it

[the remark this answers is not in the document]
Comment (expressvpn-tom-l) [reply windows/wolfssl-user_settings-64.h:12]: Got it, thanks! However, I'd suggest make this a separate PR instead of burying it in a "Refactoring" PR.

**Review by expressvpn-tom-l — APPROVED** [review]
LGTM

**Pull request merged**